### ISSUE TITLE
[a11y] 11.4.1.2 Name-role value 

### ISSUE THREAD

**Issue by tschiebel** [owncloud/ios-app#1341]
audit: https://infinite.owncloud.com/s/qSvYucPltEYUhhc

https://github.com/BIK-BITV/BIK-App-Test/tree/main/Pr%C3%BCfschritte/de
https://www.etsi.org/deliver/etsi_en/301500_301599/301549/03.02.01_60/en_301549v030201p.pdf

### Personal Space

- [x] When the elements on the page are sorted, the status of the sorting is
sorting is only shown visually by arrows. However, this visual representation
a limitation for non-visual users, however, as they do not receive any
information about the current sorting.
![image](https://github.com/owncloud/ios-app/assets/132444655/1aae3aac-9a7a-4c84-9712-f5db32e4b56e)

> **Fix:** the VoiceOver description for the button and its menu now includes the direction.

- [x] Both directory and file names are buttons and can be activated.
be activated. Unfortunately, these elements lack a corresponding role to
their function as buttons appropriately.

> **Fix**: cells representing files and folders have been consolidated for VoiceOver, presenting as a single element with a single accessibility description and proper role.

- [x] Although the "Start multiple selection" button has an accessible name
name, but this always remains the same, regardless of whether the multiple
multiple selection has been activated or not. It would be advisable
either change the accessible name or display the status of the multiple
multiple selection status.

> **Fix:**: the button now has the correct description depending on whether multi selection is active or not

- [x] In multiple selection mode, further buttons appear at the bottom of the page.
buttons appear at the bottom of the page. Unfortunately, these buttons have no role and there is no hint explaining their function.
a hint explaining their function.
![image](https://github.com/owncloud/ios-app/assets/132444655/cf781595-f0c9-4738-9080-609b22a15145)

> **Fix:**: the buttons are now properly defined accessibility elements with a button role.

- [x] Both date and time information are displayed unformatted and
unformatted and incomprehensible. Further details can be found at
https://a11y-guidelines.orange.com/en/mobile/ios/development/#date-timeand-numbers.
![image](https://github.com/owncloud/ios-app/assets/132444655/48bfb2af-fc11-4e9d-b648-b5cf39cf59cb)
![image](https://github.com/owncloud/ios-app/assets/132444655/3fafa6b2-140f-40c9-beda-a43773d0f30e)

> **Fix**: dates included in accessibility labels for items are now formatted in a way so they make sense when being read by VoiceOver
 
### Menu

- [x] The account URL appears to be a button, but it has not been implemented as such
with a corresponding role.
![image](https://github.com/owncloud/ios-app/assets/132444655/0ac8a32a-c782-4630-b8bd-5ee714badc27)

> **Fix:**: the cell now has a consolidated accessibility description and the correct role

- [x] The "Personal" button has no role.

> **Fix**: the `button` role has been added. 

- [x] All submenus, such as "Shared with me", have no role.
![image](https://github.com/owncloud/ios-app/assets/132444655/59987620-b05e-4e55-bb28-668e2c78a297)

> **Fix**: the sidebar items now have a proper role.

- [x] The "Spaces" button does not have a role and informs users that it is a
it is a collapsible element, but this is not the case.
![image](https://github.com/owncloud/ios-app/assets/132444655/c985470d-ed82-43b9-9276-155cb385c34a)

> **Fix**: the item has a role and is indeed expandable and collapsible. For items to show up, however, the account needs to be member of at least one space.

### Shared with me 

- [x] Both date and time information are output unformatted and
incomprehensible - see comment on page 2.

> **Fix:** No longer applicable testing with version 12.2: items are presented differently and no longer include that metadata.

- [x] Buttons that are located in the "Accepted" area, such as directory names
have no role.
![image](https://github.com/owncloud/ios-app/assets/132444655/440110c7-d15d-490c-95fe-af38a126449f)

> **Fix**: cells representing files and folders have been consolidated for VoiceOver, presenting as a single element with a single accessibility description and proper role.

### Spaces

- [x] The Space buttons do not have an assigned role and the
hint is also missing.

Note: A possible hint could be, for example, a descriptive text
that explains the purpose or function of the buttons. In this way
this way, users with visual impairments receive additional information to better
better understand and use the buttons.

 > **Fix**: space cells now have a role, the name of the space is presented as label, the subtitle as hint and the cells offer additional actions via accessibility custom actions.

**Comment by jesmrec** [comment]
This one is done
